Modules/_asynciomodule.c

```diff
@@ -2452,12 +2372,113 @@ module_init(void)
 
 PyDoc_STRVAR(module_doc, "Accelerator module for asyncio");
 
+
+/*[clinic input]
+_asyncio._register_task
+
+    loop: object
+    task: object
+
+Register a new task in asyncio as executed by loop.
+
+Returns None.
+[clinic start generated code]*/
+
+static PyObject *
+_asyncio__register_task_impl(PyObject *module, PyObject *loop,
+                             PyObject *task)
+/*[clinic end generated code: output=54c5cb733dbe0f38 input=9b5fee38fcb2c288]*/
+{
+    if (PyObject_SetItem(asyncio__all_tasks, task, loop) < 0) {
+        return NULL;
+    }
+    Py_RETURN_NONE;
+}
+
+
+/*[clinic input]
+_asyncio._enter_task
+
+    loop: object
+    task: object
+
+Enter into task execution or resume suspended task.
+
+Task belongs to loop.
+
+Returns None.
+[clinic start generated code]*/
+
+static PyObject *
+_asyncio__enter_task_impl(PyObject *module, PyObject *loop, PyObject *task)
+/*[clinic end generated code: output=a22611c858035b73 input=de1b06dca70d8737]*/
```

Comment:
I prefer not to use `_impl` functions in C code.  Better to implement `int enter_task(loop, task)` function and use it in `Task` implementation directly.  Then you call that function in `_asyncio__enter_task_impl` and return `NULL`/`None` based on its return value.  This way you also don't need to decref `None` in `Task` code, and the code is overall cleaner.

Reply:
Ok